src/Quahog/Client.php

```diff
@@ -257,14 +261,34 @@ private function _sendCommand($command)
     /**
      * A wrapper to cleanly read a response from clamd.
      *
+     * @param bool $removeId
+     * @param string $readUntil
      * @return string
+     * @throws ConnectionException
      */
-    private function _receiveResponse()
+    private function _receiveResponse($removeId = false, $readUntil = "\n")
     {
-        $result = $this->_socket->read(4096);
+        $result = null;
+        $readUntilLen = strlen($readUntil);
+        do {
+            if ($this->_socket->selectRead($this->_timeout)) {
+                $rt = $this->_socket->read(4096, PHP_NORMAL_READ);
+                if ($rt == "") {
+                    break;
+                }
+                $result .= $rt;
+                if (strcmp(substr($result, 0 - $readUntilLen), $readUntil) == 0) {
+                    break;
+                }
+            } else {
+                throw new ConnectionException("Timeout waiting to read response");
```

Comment:
I agree with @SenseException . This coupled with the PHP_NORMAL_READ / PHP_BINARY_READ feels like a bc break to me too. Not really sure in which scenario the behaviour would really change though (other than throwing an exception instead of an implicit connection drop). I think the cleanest way would be to make it configurable. Maybe just add another construct parameter for the read-mode and change the code so it uses the old implicit connection drop way if no timeout is passed. Something like that would be really nice. 
Otherwise i would suggest a major version bump.